python_modules/libraries/dagster-k8s/dagster_k8s/ops/k8s_job_op.py

```diff
@@ -254,6 +254,10 @@ def execute_k8s_job(
 
     job_name = get_k8s_job_name(context.run_id, context.op.name)
 
+    retry_number = context.retry_number
+    if retry_number > 0:
+        job_name = f"{job_name}-{retry_number}"
```

Comment:
does `get_k8s_job_name` truncate down to the limit such that adding `-N` would exceed or does it leave some buffer?

Reply:
the old name was always 33 characters (one of the reasons the name is so bad is that it just hexdigests it) - so as long as we don't have ~10^20 retries on a single op we're good